implementations/rust/ockam/ockam_api/src/authority_node/authority.rs

```diff
@@ -334,7 +345,331 @@ impl Authority {
+    use std::str::FromStr;
+    use std::time::Duration;
+
+    /// This test gets a reference to the postgres database and starts 2 authority nodes
+    /// to make sure that they can work even when using the same database.
+    ///
+    /// We test:
+    ///
+    ///  - That adding a member works with trusted identities
+    ///  - Issuing a credential works
+    ///  - Issuing and accepting a token works
+    #[test]
+    fn test_create_two_authority_managed_nodes_using_the_same_postgres_database() {
+        let result = execute_test(|db_base, ctx1_base, ctx2_base, ctx_client_base| {
+            let db = db_base.clone();
+            let ctx1 = ctx1_base.try_clone().unwrap();
+            let ctx2 = ctx2_base.try_clone().unwrap();
+            let ctx_client = ctx_client_base.try_clone().unwrap();
+            async move {
+                let port1 = random_port();
+                let port2 = random_port();
+                let secure_channels: Arc<SecureChannels> = secure_channels().await?;
+                let identities_creation = secure_channels.identities().identities_creation();
+
+                let enroller1 = identities_creation.create_identity().await?;
+                let enroller2 = identities_creation.create_identity().await?;
+
+                let authority1 = start_authority_node(
+                    db.clone(),
+                    &ctx1,
+                    port1,
+                    "authority-node-1",
+                    &[enroller1.clone()],
+                )
+                .await?;
+                let authority2 = start_authority_node(
+                    db,
+                    &ctx2,
+                    port2,
+                    "authority-node-2",
+                    &[enroller2.clone()],
+                )
+                .await?;
+
+                let client1 = make_authority_node_client(
+                    &ctx_client,
+                    secure_channels.clone(),
+                    &authority1.identifier,
+                    &MultiAddr::from_str(&format!("/dnsaddr/127.0.0.1/tcp/{}/secure/api", port1))?,
+                    &enroller1,
+                )
+                .await?;
+                let client2 = make_authority_node_client(
+                    &ctx_client,
+                    secure_channels.clone(),
+                    &authority2.identifier,
+                    &MultiAddr::from_str(&format!("/dnsaddr/127.0.0.1/tcp/{}/secure/api", port2))?,
+                    &enroller2,
+                )
+                .await?;
+
+                // adding members must work for both authorities
+                let identities_creation = secure_channels.identities().identities_creation();
+                let member1 = identities_creation.create_identity().await?;
+                let member2 = identities_creation.create_identity().await?;
+
+                add_member(&ctx_client, &client1, &member1).await?;
+                add_member(&ctx_client, &client2, &member2).await?;
```

Comment:
What's the point of doing the same operations with different members on different clients? I'd also add cases to prove the two clients are isolated by adding the same member to both of them and asserting that you can't add the same member twice in a given client.

Reply:
The purpose of the test is show that the authority nodes that are sharing the same database are able to respond to their own clients. 

> I'd also add cases to prove the two clients are isolated by adding the same member to both of them and asserting that you can't add the same member twice in a given client.

Good idea, I'm going to add this case.

Reply:
> sharing the same database are able to respond to their own clients.

Ok, I got it now, thanks

Reply:
> you can't add the same member twice in a given client.

Hmm, actually we can, this just updates the member attributes. I pushed some changes to the test showing that updating a member works ok.e